### PR TITLE
[on-chain config] register valdiator set into ON_CHAIN_CONFIG_REGISTRY

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

1. setup correct flow of publishing new config
2. break on_chain_config into multiple files
3. modify the access path and register validator set

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: types/src/on_chain_config/mod.rs [hunk 1]
```diff
@@ -60,8 +73,9 @@ pub trait ConfigStorage {
 /// Trait to be implemented by a Rust struct representation of an on-chain config
 /// that is stored in storage as a deserialized byte array
 pub trait OnChainConfig: Send + Sync + DeserializeOwned {
+    const ADDRESS: &'static str = "0xA550C18";
```
Comment (christinacelee) [inline types/src/on_chain_config/mod.rs:76]: maybe comment on what this value is? Or ideally, can we define this value as a constant somewhere and use that?

Comment (zekun000) [reply types/src/on_chain_config/mod.rs:76]: there's association_address() but it's not a const fn

**Comment by christinacelee** [comment]
LGTM on Rust-side! will let @runtian-zhou take a look at Move-side

File: language/stdlib/modules/libra_configs.move [hunk 2]
```diff
@@ -50,12 +50,11 @@ module LibraConfig {
     }
 
     // Publish a new config item to a new value and trigger a reconfiguration.
-    public fun publish_new_config<Config: copyable>(payload: Config) acquires Configuration {
+    public fun publish_new_config<Config: copyable>(payload: Config) {
         // TODO: impose proper permission checks
         // Callable by the any address for now.
 
         move_to_sender(T{ payload });
-        reconfigure_();
```
Comment (runtian-zhou) [inline language/stdlib/modules/libra_configs.move:58]: Maybe some comments here? Publishing a new configuration will not trigger a reconfiguration immediately. Instead we need to wait all of the validators to start to use this config before triggering a reconfiguration change.

**Pull request closed without merging**